### PR TITLE
Fix: Substitute concrete path version into route templates

### PR DESCRIPTION
**Pull request by abeljim8am**
## Summary

Routes mounted with `version "v1", using: :path` emit OAS paths with the
literal `{version}` placeholder instead of the version string.

### Before
```ruby
class API < Grape::API
  version "v1", using: :path
  get "items" do; end
end
```
```yaml
paths:
  /{version}/items: ...
```

### After
```yaml
paths:
  /v1/items: ...
```

## What changed

`sanitize_path` now substitutes the route's concrete version into both
`{version}` and `:version` segments. Routes without a concrete path
version are unchanged.

### CODE REVIEW

File: lib/grape_oas/api_model_builders/path.rb [hunk 1]
```diff
@@ -95,10 +95,25 @@ def build_operation(route, path_param_name_map: nil, template_override: nil)
...
+      def path_version_segment?(path)
+        path.start_with?("/:version/", "/:version(") || path == "/:version"
+      end
+
+      def concrete_path_version(version)
+        values = Array(version)
+        return nil unless values.length == 1
+
+        values.first.to_s
       end
```
Comment (numbata) [inline lib/grape_oas/api_model_builders/path.rb:124]: Empty or blank version string would produce a double-slash path (e.g. //items).

Maybe add a guard like this?
```ruby
def concrete_path_version(version)
  values = Array(version)
  return nil unless values.length == 1

  value = values.first.to_s
  value.empty? ? nil : value
end
```

wdyt?

File: test/grape_oas/api_model_builders/path_test.rb [hunk 2]
```diff
@@ -119,6 +119,89 @@ def test_handles_nested_path_parameters
...
+          end
+        end
+
+        builder = Path.new(api: @api, routes: api_class.routes)
+        builder.build
+
+        path = @api.paths.first
+
+        assert_equal "/v1/items/{id}", path.template
+      end
```
Comment (numbata) [inline test/grape_oas/api_model_builders/path_test.rb:203]: `filtered_by_namespace?` now compares against version-substituted paths — e.g. `/v1/items` instead of `/{version}/items`. This changes what the namespace filter matches against. Could you add a test with something like `version "v1", using: :path` combined with a `namespace_filter` to confirm filtering still works as expected?

**Review by numbata — COMMENTED** [review]
Thanks for the clean fix, @abeljim8am! I left a couple of comments — one edge case to guard against and a missing test for the namespace filter interaction. Should be quick to address.

File: test/grape_oas/api_model_builders/path_test.rb (hunk 2 above)
Comment (abeljim8am) [reply test/grape_oas/api_model_builders/path_test.rb:203]: Sounds good, I will add the test.

File: lib/grape_oas/api_model_builders/path.rb (hunk 1 above)
Comment (abeljim8am) [reply lib/grape_oas/api_model_builders/path.rb:124]: fixed
